code/modules/power/apc.dm

```diff
@@ -716,7 +716,43 @@
 	if(istype(user,/mob/living/carbon/human))
 		var/mob/living/carbon/human/H = user
 
-		if(H.species.can_shred(H))
+		if(isipc(H) && H.a_intent == I_GRAB)
+			if(emagged || stat & BROKEN)
+				var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
+				s.set_up(3, 1, src)
+				s.start()
+				H << "<span class='danger'>The APC power currents surge eratically, damaging your chassis!</span>"
+				H.adjustFireLoss(10, 0)
+			else if(src.cell && src.cell.charge > 0)
+				if(H.nutrition < 450)
+					if(src.cell.charge >= 500)
+						H.nutrition += 50
+						src.cell.charge -= 500
+					else
+						H.nutrition += src.cell.charge / 10
+						src.cell.charge = 0
+
+					user << "<span class='notice'>You slot your fingers into the APC interface and siphon off some of the stored charge for your own use.</span>"
+
+					if (src.cell.charge < 0)
+						src.cell.charge = 0
+					if (H.nutrition > 500)
+						H.nutrition = 500
+					if (prob(1))

```

Comment:
I'm still of the opinion that 1% is too often for this. I'd say make it a rarer but more severe thing